main/modes/mode_paint.h

```diff
@@ -0,0 +1,86 @@
+#if !defined(MODE_PAINT_H_)
+#define _MODE_PAINT_H_
+
+#include "swadgeMode.h"
+
+// The number of colors in the palette and the max number of colors an image can be saved with
+#define PAINT_MAX_COLORS 16
+
+
+// The total number of save slots available
+#define PAINT_SAVE_SLOTS 8
+
+// Whether to have the LEDs show the current colors
+#define PAINT_ENABLE_LEDS (0x0001 << (PAINT_SAVE_SLOTS * 2))
+
+// Whether to play SFX on drawing, etc.
+#define PAINT_ENABLE_SFX  (0x0002 << (PAINT_SAVE_SLOTS * 2))
+
+// Whether to play background music
+#define PAINT_ENABLE_BGM  (0x0004 << (PAINT_SAVE_SLOTS * 2))
+
+// Default to LEDs, SFX, and music on, with slot 0 marked as most recent
+#define PAINT_DEFAULTS (PAINT_ENABLE_LEDS | PAINT_ENABLE_SFX | PAINT_ENABLE_BGM)
+
+// Mask for the index that includes everything except the most-recent index
+#define PAINT_MASK_NOT_RECENT (PAINT_ENABLE_LEDS | PAINT_ENABLE_SFX | PAINT_ENABLE_BGM | ((1 << PAINT_SAVE_SLOTS) - 1))
+
+// The size of the buffer for loading/saving the image. Each chunk is saved as a separate blob in NVS
+#define PAINT_SAVE_CHUNK_SIZE 1024
+
+
+#define TFT_WIDTH 280
```

Comment:
Ah, I think I did this because `display->h` and `display->w` weren't defined at the point where I was defining `PAINT_CANVAS_OFFSET_{X,Y}`, but that's totally not how #defines work so I can get rid of this.